src/d3a/d3a_core/sim_results/endpoint_buffer.py

```diff
@@ -45,19 +45,22 @@ def __init__(self, job_id, initial_params, area, export_plots=True):
             "elapsed_time_seconds": 0,
             "percentage_completed": 0
         }
+        self.should_export_plots = should_export_plots
+
         self.market_unmatched_loads = MarketUnmatchedLoads(area)
-        self.price_energy_day = MarketPriceEnergyDay()
+        self.price_energy_day = MarketPriceEnergyDay(should_export_plots)
         self.market_bills = MarketEnergyBills()
         self.cumulative_bills = CumulativeBills()
         self.balancing_bills = MarketEnergyBills(is_spot_market=False)
```

Comment:
Try to skip initialising of results related to balancing markets when simulation is executed without balancing market.